browser/tor/tor_profile_service_impl.cc

```diff
@@ -46,42 +46,44 @@ void TorProfileServiceImpl::ReLaunchTor(const TorConfig& config) {
   tor_launcher_factory_->ReLaunchTorProcess(config);
 }
 
-void TorProfileServiceImpl::SetNewTorCircuitOnIOThread(
-    const scoped_refptr<net::URLRequestContextGetter>& getter,
-    std::string host) {
-  DCHECK_CURRENTLY_ON(BrowserThread::IO);
-  const TorConfig tor_config = tor_launcher_factory_->GetTorConfig();
-  if (tor_config.empty())
-    return;
-  auto* proxy_resolution_service =
-    getter->GetURLRequestContext()->proxy_resolution_service();
-  DCHECK(proxy_resolution_service);
-  TorProxyConfigService::TorSetProxy(proxy_resolution_service,
-                                     tor_config.proxy_string(),
-                                     host,
-                                     &tor_proxy_map_,
-                                     true);
+void TorProfileServiceImpl::OnSetNewTorCircuitComplete(bool success) {
+  if (tor_circuit_callback_)
+    std::move(tor_circuit_callback_).Run(success);
+}
+
+void TorProfileServiceImpl::OnProxyLookupComplete(
+    int32_t net_error,
+    const base::Optional<net::ProxyInfo>& proxy_info) {
+  bool success = proxy_info.has_value() && !proxy_info->is_direct();
+  if (tor_circuit_callback_) {
+    std::move(tor_circuit_callback_).Run(success);
+    binding_.Close();
```

Comment:
shouldn't we close the binding in any case?

Reply:
fixed in 58622cb17913fb4e4925f1d2943f2d561a4bdbd4